File: microservices/find/models/Category.js
const mongoose = require("mongoose");
const Schema = mongoose.Schema;


const categorySchema = Schema({
    iso:{type: String, required: true},
    sort:{type: Number, required: false},
    name:{type: String, required: true},
    display_name:{type: [], required: true},
    system_key:{type: String, required: true},
    slug:{type: String, required: true},
    description:{type: String, required: false},
    media:{type: []},
    published:{type: Boolean, required: true},
    checked:{type: Boolean, required: false},
    created_at:{type: Date, required: true},
    additional:{type: String, required: false},

});


module.exports = Category = mongoose.model("categories", categorySchema);

File: plugins/v2importer/models/category.py
from database.db import db
from flask_mongoengine import BaseQuerySet
import datetime
from slugify import slugify, Slugify, UniqueSlugify


class Category(db.Document):
    iso = db.StringField(required=True)
    sort = db.IntField(required=False, default=0)
    name = db.StringField(required=True, unique=True)
    sku = db.StringField(required=True)
    slug = db.StringField(required=True, unique=True, default=None)
    description = db.StringField(required=False, default='')
    media = db.ListField()
    attributes = db.ListField()
    published = db.BooleanField(default=True)
    checked = db.BooleanField(default=False)
    created_at = db.DateTimeField(default=datetime.datetime.now, required=True)
    meta = {
        'indexes': ['-created_at', 'slug', 'name'],
        'ordering': ['-created_at', 'name'],
        'collection': 'categories',
        'queryset_class': BaseQuerySet
    }

    def __setattr__(self, key, value):
        super(Category, self).__setattr__(key, value)
        if key == 'name':
            self.slug = slugify(self.name, to_lower=True)
